File: src/main/java/de/blazemcworld/fireflow/command/PlayCommand.java
package de.blazemcworld.fireflow.command;

import de.blazemcworld.fireflow.space.Space;
import de.blazemcworld.fireflow.space.SpaceManager;
import de.blazemcworld.fireflow.util.Transfer;
import de.blazemcworld.fireflow.util.Translations;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.format.NamedTextColor;
import net.minestom.server.command.builder.Command;
import net.minestom.server.entity.Player;

public class PlayCommand extends Command {

    public PlayCommand() {
        super("play");

        setDefaultExecutor((sender, ctx) -> {
            if (sender instanceof Player player) {
                Space space = SpaceManager.getSpaceForPlayer(player);
                if (space == null) {
                    sender.sendMessage(Component.text(Translations.get("error.needs.space")).color(NamedTextColor.RED));
                    return;
                }

                if (space.play == player.getInstance()) {
                    sender.sendMessage(Component.text(Translations.get("error.already.playing")).color(NamedTextColor.RED));
                    return;
                }

                Transfer.move(player, space.play);
            } else {
                sender.sendMessage(Component.text(Translations.get("error.needs.player")).color(NamedTextColor.RED));
            }
        });
    }

}

File: src/main/kotlin/de/blazemcworld/fireflow/command/PlayCommand.kt
package de.blazemcworld.fireflow.command

import de.blazemcworld.fireflow.space.SpaceManager
import de.blazemcworld.fireflow.util.sendError
import net.minestom.server.command.builder.Command
import net.minestom.server.command.builder.CommandExecutor
import net.minestom.server.coordinate.Pos
import net.minestom.server.entity.Player

object PlayCommand : Command("play") {
    init {
        defaultExecutor = CommandExecutor exec@{ sender, _ ->
            if (sender !is Player) return@exec

            val space = SpaceManager.currentSpace(sender)

            if (space == null) {
                sender.sendError("You need to be in a space to do this!")
                return@exec
            }
            if (space.playInstance == sender.instance) {
                sender.sendError("You are already playing on this space!")
                return@exec
            }
            sender.setInstance(space.playInstance, Pos.ZERO)
        }
    }
}